From 7ee375f9f394ab68ab496fef48e30abc05df63a2 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Wed, 12 May 2021 11:35:22 +0200
Subject: [PATCH] arm64: dts: ipq8074: Use GIC MSI for PCI

Instead of using a single hardcoded MSI interrupt,
switch to using GIC MSI.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 arch/arm64/boot/dts/qcom/ipq8074.dtsi | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

--- a/arch/arm64/boot/dts/qcom/ipq8074.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq8074.dtsi
@@ -624,6 +624,13 @@
 			interrupt-controller;
 			#interrupt-cells = <0x3>;
 			reg = <0x0b000000 0x1000>, <0x0b002000 0x1000>;
+			ranges = <0 0xb00a000 0xffd>;
+
+			v2m0: v2m@0 {
+				compatible = "arm,gic-v2m-frame";
+				msi-controller;
+				reg = <0x0 0xffd>;
+			};
 		};
 
 		timer {
@@ -723,8 +730,6 @@
 				 <0x82000000 0 0x10220000 0x10220000
 				  0 0xfde0000>; /* non-prefetchable memory */
 
-			interrupts = <GIC_SPI 85 IRQ_TYPE_LEVEL_HIGH>;
-			interrupt-names = "msi";
 			#interrupt-cells = <1>;
 			interrupt-map-mask = <0 0 0 0x7>;
 			interrupt-map = <0 0 0 1 &intc 0 142
@@ -760,6 +765,8 @@
 				      "axi_s",
 				      "ahb",
 				      "axi_m_sticky";
+			msi-parent = <&v2m0>;
+
 			status = "disabled";
 		};
 
@@ -786,8 +793,6 @@
 				 <0x82000000 0 0x20220000 0x20220000
 				  0 0xfde0000>; /* non-prefetchable memory */
 
-			interrupts = <GIC_SPI 52 IRQ_TYPE_LEVEL_HIGH>;
-			interrupt-names = "msi";
 			#interrupt-cells = <1>;
 			interrupt-map-mask = <0 0 0 0x7>;
 			interrupt-map = <0 0 0 1 &intc 0 75
@@ -830,6 +835,8 @@
 				      "ahb",
 				      "axi_m_sticky",
 				      "axi_s_sticky";
+
+			msi-parent = <&v2m0>;
 			status = "disabled";
 		};
 
